src/katgpucbf/xbgpu/correlation.py

```diff
@@ -73,13 +73,20 @@ class CorrelationTemplate:
         The number of time samples to be processed per frequency channel.
     """
 
-    def __init__(self, context: AbstractContext, n_ants: int, n_channels: int, n_spectra_per_heap: int) -> None:
+    def __init__(
+        self,
+        context: AbstractContext,
+        n_ants: int,
+        n_channels: int,
+        n_spectra_per_heap: int,
+        sample_bitwidth: int,
```

Comment:
I'd rather call this `input_sample_bits` for consistency with fgpu. I'm not too fussed about the `input_` prefix (although it does make it clearer, because the output is 32-bit), but definitely `bits` rather than `bitwidth` for consistency.